src/plugins/unified_histogram/public/chart/histogram.tsx

```diff
@@ -141,11 +138,13 @@ export function Histogram({
 
       const adapterTables = adapters?.tables?.tables;
       const totalHits = computeTotalHits(hasLensSuggestions, adapterTables, isPlainRecord);
+      const totalHitsStatus = isLoading
+        ? UnifiedHistogramFetchStatus.loading
+        : response?._shards?.failed || response?.timed_out
+        ? UnifiedHistogramFetchStatus.error
+        : UnifiedHistogramFetchStatus.complete;
 
-      onTotalHitsChange?.(
-        isLoading ? UnifiedHistogramFetchStatus.loading : UnifiedHistogramFetchStatus.complete,
-        totalHits ?? hits?.total
```

Comment:
This line might be creating another set of problems when it uses the previous values as a new one.

Reply:
I think it's time we do the refactoring https://github.com/elastic/kibana/issues/177156

Reply:
I definitely agree this logic has become a mess, but do you have ideas on the right approach for #177156? Adding `track_total_hits` to the main request would hurt performance for large indices, and otherwise it seems we'd have to add a third `_count` request to each fetch.

Reply:
Yeah, we could experiment with adding a `_count` request. At least it would be triggered by Discover directly and it would not depend on whether the chart is shown or not.

Reply:
That's true, it would definitely be cleaner to maintain at least. Let's discuss it with the team, but I agree it could make sense to experiment with it and assess performance impact.

Reply:
yes, we should aim to refactor this. Ideally we shouldn't add another request, or sync with ES folks first, if we want to. or maybe make the exact number request optional in the UI for users that want to know exactly .. .something like this. If currently not possible to use Lens as a embeddable that just renderes given data as a property, and integrate the Lens data fetching into Discover?

Reply:
>If currently not possible to use Lens as a embeddable that just renderes given data as a property, and integrate the Lens data fetching into Discover?

I think we do this in a very specific case currently (ES|QL mode with no transformational commands?), but I don't believe it's generally supported. Even just having the ability to control when Lens fetches occur would go a long way in simplifying our data fetching and Unified Histogram overall.

But I also don't think passing fetched data to Lens or controlling when fetching occurs would help us in situations like these, since the Lens request actually failed due to aggregations and the total hits may be inaccurate/unavailable. To avoid this and ensure total hits are always retuned whenever the documents request succeeds, we'd likely need to use something like `track_total_hits` or a second duplicate search request just to get the count.

It might be ok for total hits to partially fail in these situations though as long as the table is rendered (like it is with this fix), since I think we can safely assume histogram request failures are an edge case. So it feels like architecture would be the primary driver for changes here.